src/sentry/tasks/split_discover_dataset.py

```diff
@@ -0,0 +1,226 @@
+import datetime
+import logging
+
+import sentry_sdk
+from celery.exceptions import SoftTimeLimitExceeded
+
+from sentry import options
+from sentry.models.dashboard_widget import (
+    DashboardWidget,
+    DashboardWidgetQuery,
+    DashboardWidgetTypes,
+)
+from sentry.search.events.builder import QueryBuilder
+from sentry.search.events.types import ParamsType, QueryBuilderConfig
+from sentry.snuba.dataset import Dataset
+from sentry.snuba.referrer import Referrer
+from sentry.tasks.base import instrumented_task
+from sentry.utils import metrics
+from sentry.utils.cache import cache
+from sentry.utils.query import RangeQuerySetWrapper
+
+logger = logging.getLogger("sentry.tasks.split_discover_dataset")
+
+TASK_QUERY_PERIOD = "30m"
+DASHBOARD_QUERY_PERIOD = "1h"
+
+
+def _get_widget_processing_batch_key() -> str:
+    return "on-demand-metrics:widgets:currently-processing-batch"
```

Comment:
```suggestion
    return "split-discover-dataset:widgets:currently-processing-batch"
```
This can't be the same string otherwise it'll conflict w/ the on_demand_task